includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -955,11 +966,25 @@ private function get_stylesheet_priority( DOMNode $node ) {
 			if ( 'print' === $node->getAttribute( 'media' ) ) {
 				$priority += $print_priority_base;
 			}
-		} elseif ( $node instanceof DOMElement && 'style' === $node->nodeName ) {
-			$element_id = (string) $node->getAttribute( 'id' );
-			if ( 'admin-bar-inline-css' === $element_id ) {
+		} elseif ( $node instanceof DOMElement && 'style' === $node->nodeName && $node->hasAttribute( 'id' ) ) {
+			$id = $node->getAttribute( 'id' );
+			if (
+				preg_match( '/^(?<handle>.+)-inline-css$/', $id, $matches ) &&
+				wp_style_is( $matches['handle'], 'registered' ) &&
+				0 === strpos( wp_styles()->registered[ $matches['handle'] ]->src, get_template_directory_uri() )
+			) {
+				// Parent theme inline style.
+				$priority = 2;
+			} elseif (
+				preg_match( '/^(?<handle>.+)-inline-css$/', $id, $matches ) &&
```

Comment:
Minor points: 

1. The `preg_match()` call with its `$matches` variable, and the `wp_style_is( $matches['handle'], 'registered' )` call as well could be factored out of the conditions and then used in both
2. The second condition should include a check for `get_stylesheet() !== get_template()` for strict correctness.

Reply:
Resolved via https://github.com/ampproject/amp-wp/pull/3877/commits/93c2760e4176bf1b159cafd04d6d42ae520996a6